code/core/src/instrumenter/instrumenter.ts

```diff
@@ -263,6 +255,24 @@ export class Instrumenter {
     }
   }
 
+  loadParentWindowState = () => {
+    try {
+      this.state = global.window?.parent?.__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__ || {};
```

Comment:
This should be changed to use `window.postMessage()` to enable safe cross-origin communication.